### ISSUE TITLE
Installation issue: spla-1.4.0 fails to build with amdblis

### ISSUE THREAD

**Issue by dev-zero** [LLNL/spack#24357]
### Steps to reproduce the issue

```console
$ spack install sirius@master%gcc@10.2.0 +fortran ^amdfftw+openmp ^amdblis threads=openmp ^amdlibflame +debug threads=openmp
==> Warning: Missing a source id for sirius@master
[+] /usr (external python-3.6.12-oynjtia5giosmdmxg6dp4loonpke3ofa)
[+] /opt/cray/pe/mpich/8.1.4/ofi/gnu/9.1 (external cray-mpich-8.1.4-ozxf5hnn3zeqyqa5wzcszm64xxbnpbwd)
[+] /usr (external texinfo-6.5-aur2eqnugwmvldb5dhcwx34pwttf7j2f)
==> cmake@3.20.1 : has external module in ['CMake/3.20.1']
[+] /apps/eiger/UES/jenkins/1.4.0/software/CMake/3.20.1 (external cmake-3.20.1-hmpltm5p4jmjjd7aqvkzx56kilzvbdvs)
[+] /project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/gsl-2.6-ztx3zh4eeqsuekqz72iymijizseyq2fa
[+] /project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/zlib-1.2.11-okpkimphjy2o77ldaznr2zpkfotqvw6f
[+] /usr (external perl-5.26.1-6wlidbl53inaklldlwdml7eieh4gbrz4)
[+] /usr (external pkg-config-0.29.2-qtj2nqd4r4g7brmosr7hgbq55pogpfnx)
[+] /project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/amdblis-3.0-lu2dyperyf3wntmvghzigdwhwcbbxrxw
[+] /project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/amdfftw-3.0-jvrhyuzo6smqo2akda6ziushw76fxh6y
[+] /project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/spglib-1.16.1-4zvs7o7ymbhfzrvgsmddjeulvjfy3ses
[+] /project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/hdf5-1.10.7-ot4nlc5h2crp6tuapwzenw5nkqg2r5nv
[+] /project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/libxc-5.1.3-zpto2srxgvqbrszizstc5xj3i65qdd4j
==> Installing spla-1.4.0-2jfv2mrs7ixl5hhygmutr57lpnloc7nm
==> No binary for spla-1.4.0-2jfv2mrs7ixl5hhygmutr57lpnloc7nm found: installing from source
==> Using cached archive: /project/d110/timuel/spack/var/spack/cache/_source-cache/archive/36/364a9fe759fddec8a0839cf79f1cf0619fc36f4d4c15f1c2b1f437249d7840c6.tar.gz
==> No patches needed for spla
==> spla: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/spla-1.4.0-2jfv2mrs7ixl5hhygmutr57lpnloc7nm' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/spla-1.4.0-2jfv2mrs7ixl5hhygmutr57lpnloc7nm/lib;/project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/spla-1.4.0-2jfv2mrs7ixl5hhygmutr57lpnloc7nm/lib64;/project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/amdblis-3.0-lu2dyperyf3wntmvghzigdwhwcbbxrxw/lib;/opt/cray/pe/mpich/8.1.4/ofi/gnu/9.1/lib;/opt/gcc/10.2.0/snos' '-DCMAKE_PREFIX_PATH:STRING=/project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/amdblis-3.0-lu2dyperyf3wntmvghzigdwhwcbbxrxw;/apps/eiger/UES/jenkins/1.4.0/software/CMake/3.20.1;/opt/cray/pe/mpich/8.1.4/ofi/gnu/9.1' '-DSPLA_OMP:BOOL=ON' '-DSPLA_FORTRAN:BOOL=OFF' '-DSPLA_STATIC:BOOL=OFF' '-DSPLA_GPU_BACKEND=OFF' '-DSPLA_HOST_BLAS=BLIS' '/scratch/e1000/timuel/spack-stage/spack-stage-spla-1.4.0-2jfv2mrs7ixl5hhygmutr57lpnloc7nm/spack-src'

1 error found in build log:
     12    -- Performing Test HAVE___RESTRICT__
     13    -- Performing Test HAVE___RESTRICT__ - Success
     14    -- Found MPI_CXX: /project/d110/timuel/spack/lib/spack/env/gcc/g++ (found version "3.1")
     15    -- Found MPI: TRUE (found version "3.1") found components: CXX
     16    -- Found OpenMP_CXX: -fopenmp (found version "4.5")
     17    -- Found OpenMP: TRUE (found version "4.5")
  >> 18    CMake Error at /apps/eiger/UES/jenkins/1.4.0/software/CMake/3.20.1/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
     19      Could NOT find BLIS (missing: BLIS_LIBRARIES)
     20    Call Stack (most recent call first):
     21      /apps/eiger/UES/jenkins/1.4.0/software/CMake/3.20.1/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
     22      cmake/modules/FindBLIS.cmake:74 (find_package_handle_standard_args)
     23      CMakeLists.txt:199 (find_package)
     24    

See build log for details:
  /scratch/e1000/timuel/spack-stage/spack-stage-spla-1.4.0-2jfv2mrs7ixl5hhygmutr57lpnloc7nm/spack-build-out.txt

==> Warning: Skipping build of sirius-master-a5j35gld5tf3qwba36q2lhstdaq67vgq since spla-1.4.0-2jfv2mrs7ixl5hhygmutr57lpnloc7nm failed
[+] /project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/amdlibflame-3.0-x3qa2polthampaewf7tzgr5owacon7y4
[+] /project/d110/timuel/spack/opt/spack/cray-sles15-zen2/gcc-10.2.0/spfft-1.0.3-hr52t7fhizfj34lfv6x4cya4mjplterh
==> Error: sirius-master-a5j35gld5tf3qwba36q2lhstdaq67vgq: Package was not installed
==> Error: Installation request failed.  Refer to reported errors for failing package(s).
```

### Information on your system

```
* **Spack:** 0.16.2-3156-551ae264fe
* **Python:** 3.6.12
* **Platform:** cray-sles15-zen2
* **Concretizer:** original
```

### Additional information

* [spack-build-env.txt](https://github.com/spack/spack/files/6664267/spack-build-env.txt)
* [spack-build-out.txt](https://github.com/spack/spack/files/6664269/spack-build-out.txt)

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

**Comment by dev-zero** [comment]
maybe @amd-toolchain-support would be interested?

**Comment by dev-zero** [comment]
this was fixed in #24374
